cpu/nrf52/radio/nrf802154/nrf802154.c

```diff
@@ -0,0 +1,403 @@
+    }
+    else if (txpower > -13) {
+        NRF_RADIO->TXPOWER = RADIO_TXPOWER_TXPOWER_Neg12dBm;
+    }
+    else if (txpower > -17) {
+        NRF_RADIO->TXPOWER = RADIO_TXPOWER_TXPOWER_Neg16dBm;
+    }
+    else if (txpower > -21) {
+        NRF_RADIO->TXPOWER = RADIO_TXPOWER_TXPOWER_Neg20dBm;
+    }
+    else {
+        NRF_RADIO->TXPOWER = RADIO_TXPOWER_TXPOWER_Neg40dBm;
+    }
+}
+
+static int init(netdev_t *dev)
+{
+    assert(dev);
+
+    /* initialize local variables */
+    mutex_init(&txlock);
+
+    /* reset buffer */
+    rxbuf[0] = 0;
+    txbuf[0] = 0;
+
+    /* power on peripheral */
+    power_on();
+    /* make sure the radio is disabled/stopped */
+    disable();
+
+    /* we configure it to run in IEEE802.15.4 mode */
+    NRF_RADIO->MODE = RADIO_MODE_MODE_Ieee802154_250Kbit;
+    /* and set some fitting configuration */
+    NRF_RADIO->PCNF0 = ((8 << RADIO_PCNF0_LFLEN_Pos) |
+                        (RADIO_PCNF0_PLEN_32bitZero << RADIO_PCNF0_PLEN_Pos) |
+                        (RADIO_PCNF0_CRCINC_Include << RADIO_PCNF0_CRCINC_Pos));
+    NRF_RADIO->PCNF1 = IEEE802154_FRAME_LEN_MAX;
+    /* set start frame delimiter */
+    NRF_RADIO->SFD = IEEE802154_SFD;
+    /* set MHR filters */
+    NRF_RADIO->MHRMATCHCONF = 0;
+    NRF_RADIO->MHRMATCHMAS = 0xff0007ff;
+    /* configure CRC conform to IEEE802154 */
+    NRF_RADIO->CRCCNF = ((RADIO_CRCCNF_LEN_Two << RADIO_CRCCNF_LEN_Pos) |
+                         (RADIO_CRCCNF_SKIPADDR_Ieee802154 << RADIO_CRCCNF_SKIPADDR_Pos));
+    NRF_RADIO->CRCPOLY = 0x011021;
+    NRF_RADIO->CRCINIT = 0;
+
+    /* assign default addresses */
+    luid_get(nrf802154_dev.short_addr, IEEE802154_SHORT_ADDRESS_LEN);
+    luid_get(nrf802154_dev.long_addr, IEEE802154_LONG_ADDRESS_LEN);
+
+    /* set default channel */
+    set_chan(nrf802154_dev.chan);
+
+    /* configure some shortcuts */
+    NRF_RADIO->SHORTS = RADIO_SHORTS_RXREADY_START_Msk | RADIO_SHORTS_TXREADY_START_Msk;
+
+    /* enable interrupts */
+    NVIC_EnableIRQ(RADIO_IRQn);
+    NRF_RADIO->INTENSET = RADIO_INTENSET_END_Msk;
+
+    /* switch to RX mode */
+    enable_rx();
+
+    return 0;
+}
+
+static int send(netdev_t *dev, const struct iovec *vector, unsigned count)
+{
+    (void)dev;
+
+    DEBUG("Send a packet\n");
+
+    assert(vector && count);
+
+    /* copy packet data into the transmit buffer */
+    int pos = 1;
+    for (unsigned i = 0; i < count; i++) {
+        if ((pos + vector[i].iov_len) > (IEEE802154_FRAME_LEN_MAX + 1)) {
+            DEBUG("[nrf802154] send: unable to do so, packet is too large!\n");
+            return -EOVERFLOW;
+        }
+        memcpy(&txbuf[pos], vector[i].iov_base, vector[i].iov_len);
+        pos += vector[i].iov_len;
+    }
+
+    /* specify the length of the package. */
+    txbuf[0] = pos + 1;
+
+    enable_tx();
+
+    /* trigger the actual transmission */
+    DEBUG("[nrf802154] send: putting %i byte into the ether\n", pos);
+
+    /* @todo: how about CCA, retransmits, and ACKs? */
```

Comment:
I would propose to add some comprehensive doc and a TODO list into the module description (`nrf802154.h`). Maye have look at the `* ## Implementation Status ##` section in  `gcoap.h` as example. When documented there, you can safely remove the comment above